jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileCredentialProviderFactory.kt

```diff
@@ -264,15 +321,19 @@ class ProfileCredentialProviderFactory : CredentialProviderFactory, Disposable {
         val name = this.name()
         val defaultRegion = this.properties()[ProfileProperty.REGION]
 
-        return if (this.requiresMfa(profiles)) {
-            ProfileCredentialsIdentifierMfa(name, defaultRegion)
-        } else {
-            ProfileCredentialsIdentifier(name, defaultRegion)
+        return when {
+            this.requiresMfa(profiles) -> ProfileCredentialsIdentifierMfa(name, defaultRegion)
+            this.requiresSso(profiles) -> ProfileCredentialsIdentifierSso(name, defaultRegion,
+                diskCache, this.requiredProperty(SSO_URL))
+            else -> ProfileCredentialsIdentifier(name, defaultRegion)
         }
     }
 
     private fun Profile.requiresMfa(profiles: Map<String, Profile>) = this.traverseCredentialChain(profiles)
         .any { it.propertyExists(ProfileProperty.MFA_SERIAL) }
+
+    private fun Profile.requiresSso(profiles: Map<String, Profile>) = this.traverseCredentialChain(profiles)
```

Comment:
Have you got a test case that puts SSO in a traversal chain?

Reply:
Nope, I am working on a few more unit tests